spec/users.spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.create(name: 'Mary', posts_counter: 0) }
+
+  context '#validation' do
+    it 'should accept name and posts_counter and save to the database' do
+      expect(subject).to be_valid
+    end
+
+    it 'should validate_presence_of(:name)' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'should validate_presence_of(:posts_counter)' do
+      subject.posts_counter = 0
+      expect(subject).to be_valid
+    end
+
+    it 'should validate_integer_is_positive(:posts_counter)' do
+      subject.posts_counter = -1
+      expect(subject).to_not be_valid
+    end
+
+    it 'expect posts_counter should be an integer' do
+      subject.posts_counter = 2.5
+      expect(subject).to_not be_valid
+    end
+  end
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=equal%20to%20zero.-,Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.,-Need%20a%20big), Please add a unit test for the `most_recent_posts` method in user model.

Reply:
@Omar-Muhamad done